src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -609,22 +560,13 @@ private ResourceBundle getRootResourceBundle() {
         return rootResourceBundle;
     }
 
-    private void clearCache() {
-        resourceBundleCache.clear();
+    void clearCache() {
         languageRootPaths.clear();
-
-        final List<ServiceRegistration<ResourceBundle>> regs;
-        synchronized (this) {
-            regs = new ArrayList<>(bundleServiceRegistrations.values());
-            bundleServiceRegistrations.clear();
-        }
-        for (final ServiceRegistration<ResourceBundle> serviceReg : regs) {
-            serviceReg.unregister();
-        }
+        resourceBundleRegistry.clear();
```

Comment:
Maybe `resourceBundleRegistry.unregisterAll()` would be a more descriptive name?

Reply:
Sounds good, implemented.